src/Controller/SoapController.php

```diff
@@ -22,68 +23,83 @@ class SoapController extends AbstractController
     private const DDB_WSDL_FILE = '/src/Service/MoreInfoService/Schemas/DDB/moreInfoService.wsdl';
     private const DBC_WSDL_FILE = '/src/Service/MoreInfoService/Schemas/DBC/moreInfoService.wsdl';
 
+    private $dispatcher;
+
+    /**
+     * SoapController constructor.
+     *
+     * @param EventDispatcherInterface $dispatcher
+     */
+    public function __construct(EventDispatcherInterface $dispatcher)
+    {
+        $this->dispatcher = $dispatcher;
+    }
+
     /**
      * @Route("/2.11/", name="ddb_soap")
      *
      * @param Request $request
      * @param DdbMoreInfoService $ddbMoreInfoService
-     * @param LoggerInterface $statsLogger
+     * @param StatsLoggingService $statsLoggingService
+     *    Statistics logging service
      * @param $projectDir
      *
      * @return Response
      */
-    public function ddbSoap(Request $request, DdbMoreInfoService $ddbMoreInfoService, LoggerInterface $statsLogger, $projectDir): Response
+    public function ddbSoap(Request $request, DdbMoreInfoService $ddbMoreInfoService, StatsLoggingService $statsLoggingService, $projectDir): Response
```

Comment:
The logger service implements the `LoggerInterface` this should be the type that the function takes.

Reply:
But this is autowired. Therefore it needs to be the precise service.

Reply:
I have fixed it for the private function at the bottom of the SoapController.php